diff -ur libgnomedb-3.1.2.orig/libgnomedb/Makefile.am libgnomedb-3.1.2/libgnomedb/Makefile.am
--- libgnomedb-3.1.2.orig/libgnomedb/Makefile.am	2007-10-02 14:59:41.000000000 +0200
+++ libgnomedb-3.1.2/libgnomedb/Makefile.am	2010-03-04 14:09:50.450126818 +0100
@@ -38,7 +38,6 @@
 	gnome-db-decl.h \
 	gnome-db-dsn-editor.h \
 	gnome-db-dsn-spec.h \
-	gnome-db-enum-types.h \
 	gnome-db-enums.h \
 	gnome-db-find-dialog.h \
 	gnome-db-form.h \
